client/src/store/new-request-module.ts

```diff
@@ -1885,71 +1897,74 @@ export class NewRequestModule extends VuexModule {
       this.mutateDisplayedComponent('Tabs')
     }
   }
+
+  /**
+   * Confirms whether the specified action is allowed.
+   * @param action the action to confirm
+   * @returns True if confirmed, otherwise False
+   */
   @Action
-  async confirmAction (action: string) {
+  async confirmAction (action: string): Promise<boolean> {
     try {
-      let resp = await this.getNameRequest(this.nr.id)
-      this.setNrResponse(resp)
-      if (!resp.actions.includes(action)) {
-        return false
-      }
-      return true
+      const nrData = await this.getNameRequest(this.nr.id)
+      if (!nrData) throw new Error('Got error from getNameRequest()')
+      this.setNrResponse(nrData)
+      return Boolean(nrData.actions.includes(action))
     } catch (error) {
-      if (error instanceof ApiError) {
-        await errorModule.setAppError({ id: 'get-name-requests-api-error', error: error.message } as ErrorI)
-      } else {
-        await errorModule.setAppError({ id: 'get-name-requests-error', error: error.message } as ErrorI)
-      }
-      // eslint-disable-next-line
-      console.log(error)
+      console.error('confirmAction() =', error) // eslint-disable-line no-console
+      await errorModule.setAppError(
+        { id: 'confirm-action-error', error: 'Could not confirm action' }
+      )
       return false
     }
   }
-  @Action
-  async findNameRequest () {
-    this.resetAnalyzeName()
-    this.mutateDisplayedComponent('SearchPending')
-
-    let params = {
-      nrNum: this.existingRequestSearch.nrNum,
-      phoneNumber: this.existingRequestSearch.phoneNumber,
-      emailAddress: this.existingRequestSearch.emailAddress
-    }
 
+  @Action
+  async findNameRequest (): Promise<void> {
     try {
-      let { CancelToken } = axios
-      source = CancelToken.source()
+      this.resetAnalyzeName()
+      this.mutateDisplayedComponent('SearchPending')
 
-      let resp = await axios.get('/namerequests', {
+      const params: ExistingRequestSearchI = {
+        nrNum: this.existingRequestSearch.nrNum,
+        phoneNumber: this.existingRequestSearch.phoneNumber,
+        emailAddress: this.existingRequestSearch.emailAddress
+      }
+      const { CancelToken } = axios
+      const source = CancelToken.source()
+
+      const resp = await axios.get('/namerequests', {
         params,
         cancelToken: source.token
       })
-      if (!resp.data || resp.data.length === 0) {
+
+      if (!resp || !resp.data || resp.data.length === 0) {
         this.mutateNameRequest(
           {
-            text: 'There were no records found that match the information you have entered.  Please verify the NR  ' +
-              'Number and the phone / email and try again.',
+            text: 'No records were found that match the information you entered.<br>' +
+              'Please verify the NR Number and the phone / email and try again.',
             failed: true
```

Comment:
Depending on the calling code, either:
- the calling code needs to check `if (nr?.failed)` and handle it, or
- this code should call `setAppError`